xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java

```diff
@@ -125,124 +110,35 @@ public boolean canHandleEmptyAddressListFromNameResolution() {
-
-    final String clusterName;
-
-    ClusterBalancerFactory(String clusterName) {
-      this.clusterName = clusterName;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-      if (!(o instanceof ClusterBalancerFactory)) {
-        return false;
-      }
-      ClusterBalancerFactory that = (ClusterBalancerFactory) o;
-      return clusterName.equals(that.clusterName);
-    }
-
-    @Override
-    public int hashCode() {
-      return Objects.hash(super.hashCode(), clusterName);
-    }
-
-    @Override
-    public LoadBalancer newLoadBalancer(final Helper helper) {
-      return new LoadBalancer() {
-        // Becomes non-null once handleResolvedAddresses() successfully.
-        // Assigned at most once.
-        @Nullable
-        ClusterWatcherImpl clusterWatcher;
-
-        @Override
-        public void handleNameResolutionError(Status error) {
-          if (clusterWatcher == null || clusterWatcher.edsBalancer == null) {
-            // Go into TRANSIENT_FAILURE if we have not yet received any cluster resource.
-            // Otherwise, we keep running with the data we had previously.
-            helper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(error));
-          }
-        }
-
-        @Override
-        public boolean canHandleEmptyAddressListFromNameResolution() {
-          return true;
-        }
-
-        @Override
-        public void shutdown() {
-          if (clusterWatcher != null) {
-            if (clusterWatcher.edsBalancer != null) {
-              clusterWatcher.edsBalancer.shutdown();
-            }
-            xdsClient.cancelClusterDataWatch(clusterName, clusterWatcher);
-            logger.log(
-                XdsLogLevel.INFO,
-                "Cancelled cluster watcher on {0} with xDS client {1}",
-                clusterName, xdsClient);
-          }
-        }
-
-        @Override
-        public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-          if (clusterWatcher == null) {
-            clusterWatcher = new ClusterWatcherImpl(helper, resolvedAddresses);
-            logger.log(
-                XdsLogLevel.INFO,
-                "Start cluster watcher on {0} with xDS client {1}",
-                clusterName, xdsClient);
-            xdsClient.watchClusterData(clusterName, clusterWatcher);
-          }
-        }
-      };
-    }
-  }
-
-  private static final class EdsLoadBalancingHelper extends ForwardingLoadBalancerHelper {
-    private final Helper delegate;
-    private final AtomicReference<SslContextProvider> sslContextProvider;
-
-    EdsLoadBalancingHelper(Helper helper,
-        AtomicReference<SslContextProvider> sslContextProvider) {
-      this.delegate = helper;
-      this.sslContextProvider = sslContextProvider;
-    }
+  private final class ChannelSecurityLbHelper extends ForwardingLoadBalancerHelper {
+    @Nullable
+    private SslContextProvider sslContextProvider;
+    @Nullable
+    private LoadBalancer lb;
```

Comment:
It looks strange to have `lb` field here, the whole class does not use `lb` at all. `lb` should be a field of `ChildLbState`.

Reply:
Sounds good. Changed.